src/platform_impl/web/window.rs

```diff
@@ -60,42 +59,47 @@ impl Window {
         let destroy_fn = Box::new(move || runner.notify_destroy_window(RootWI(id)));
 
         let has_focus = canvas.borrow().has_focus.clone();
-        let window = Window {
+        let inner = Inner {
             id,
+            window: window.clone(),
+            document: document.clone(),
+            canvas,
+            previous_pointer: RefCell::new("auto"),
+            register_redraw_request,
+            destroy_fn: Some(destroy_fn),
             has_focus,
-            inner: Dispatcher::new(Inner {
-                window: window.clone(),
-                document: document.clone(),
-                canvas,
-                previous_pointer: RefCell::new("auto"),
-                register_redraw_request,
-                destroy_fn: Some(destroy_fn),
-            })
-            .unwrap(),
         };
 
-        window.set_title(&attr.title);
-        window.set_maximized(attr.maximized);
-        window.set_visible(attr.visible);
-        window.set_window_icon(attr.window_icon);
+        inner.set_title(&attr.title);
+        inner.set_maximized(attr.maximized);
+        inner.set_visible(attr.visible);
+        inner.set_window_icon(attr.window_icon);
+
+        Ok(Window {
+            inner: Dispatcher::new(inner).unwrap(),
+        })
+    }
 
-        Ok(window)
+    pub(crate) fn maybe_queue_on_main(&self, f: impl FnOnce(&Inner) + Send + 'static) {
+        self.inner.dispatch(f)
+    }
+
+    // TODO: Find a way to make this non-static
+    pub(crate) fn maybe_wait_on_main<R: Send + 'static>(
```

Comment:
This should be straightforward to fix, just add some unsafe to modify the lifetime. Let me know and I can add it.

Reply:
Please do, I'm not entirely confident that I won't cause unsoundness here.

Reply:
Hm, I just realized that this will cause `MainThreadSafe::new()` and `Channel::new()` to be `unsafe`, because the `handler` could do something funny with `E`, which isn't actually `'static`.

Is it actually necessary to remove `'static` here?

Reply:
Forget I said anything.